Userland/Libraries/LibGfx/ImageFormats/PNGLoader.cpp

```diff
@@ -131,17 +132,14 @@ ErrorOr<bool> PNGImageDecoderPlugin::initialize()
     png_structp png_ptr = png_create_read_struct(PNG_LIBPNG_VER_STRING, nullptr, nullptr, nullptr);
     if (!png_ptr)
         return false;
+    AK::ScopeGuard png_cleanup = [&] { png_destroy_read_struct(&png_ptr, nullptr, nullptr); };
```

Comment:
Are we sure there’s no wonky interaction between ScopeGuard and setjmp here?
I haven’t looked closely, just commenting while doing errands so I don’t forget :)

Reply:
There is, in fact, LeakSanitizer caught it

Reply:
Removed the scope guard; no use in going near any potential UB :^)